nodelet_topic_tools/include/nodelet_topic_tools/nodelet_lazy.h

```diff
@@ -233,6 +254,33 @@ class NodeletLazy: public nodelet::Nodelet
     return pub;
   }
 
+  /** @brief
+    * Update the list of ImageTransport Publishers created by this method.
+    * It automatically reads latch boolean parameter from nh and
+    * publish topic with appropriate latch parameter.
+    *
+    * @param it ImageTransport.
+    * @param topic topic name to advertise.
+    * @param queue_size queue size for publisher.
+    * @param latch set true if latch topic publication.
+    * @return Publisher for the advertised topic.
+    */
+  image_transport::Publisher
+  advertise(image_transport::ImageTransport& it,
+            const std::string& topic, int queue_size, bool latch=false)
+  {
+    boost::mutex::scoped_lock lock(connection_mutex_);
+    image_transport::SubscriberStatusCallback connect_cb
```

Comment:
I'm reusing the `connect_cb` callback for connection and disconnection. I could change this one to declare two callbacks as in the other `advertise` method, but I'd rather change the other one to match this one, since the callback is still the same.